### ISSUE TITLE
Ability to cancel the encoder or stop it.

### ISSUE THREAD

**Issue by goxr3plus** [a-schild/jave2#18]
It's very important to add the ability to cancel the encoder , i am facing hard problems with big files 2 hours and more converting them to something other but in the middle of the long progress i decide to cancel the encoding but ... i can't .

Can we add the ability to cancel it ?

For example , let's say we use the below to convert a wav to mp3 , but the wav is 1 GB and we want to cancel it .... : 

``` JAVA 
   try {                                                            
	File source = new File("file path");		                 
	File target = new File("file path);                          
                                                                 
       //Audio Attributes                                               
	AudioAttributes audio = new AudioAttributes();               
	audio.setCodec("libmp3lame");                                
	audio.setBitRate(128000);                                    
	audio.setChannels(2);                                        
	audio.setSamplingRate(44100);                                
	                                                             
	//Encoding attributes                                        
	EncodingAttributes attrs = new EncodingAttributes();         
	attrs.setFormat("mp3");                                      
	attrs.setAudioAttributes(audio);                             
	                                                             
	//Encode                                                     
	Encoder encoder = new Encoder();                             
	encoder.encode(new MultimediaObject(source), target, attrs); 
                                                                 
   } catch (Exception ex) {                                         
	ex.printStackTrace();                                        
	succeeded = false;                                           
   }    
```

**Comment by a-schild** [comment]
Best could probably be to add a new method which runs the encoding in background and returns a CompletableFuture.

Something like this API is providing https://github.com/a-schild/nextcloud-java-api/blob/master/src/main/java/org/aarboard/nextcloud/api/utils/ConnectorCommon.java

This could then be enhanced with a cancel/abort function.

**Comment by goxr3plus** [comment]
Yes i agree , i am trying to implement it :) ,  i see the FFMpeg library is being called as a Process hm.

**Comment by a-schild** [comment]
https://stackoverflow.com/questions/6356340/killing-a-process-using-java

**Comment by a-schild** [comment]
It's implemented in the 2.4.3-SNAPSHOT release.
Just do it as usual, but start the encoding in a thread and then use the encoder.abortEncoding() method to stop the process.
It might the throw a stream closed exception, but that's normal.

       ... prepare the encoder just as usual and then start it in a thread ...
       Runnable task = () -> {
            try
            {
                encoder.encode(new MultimediaObject(source), target, attrs, listener);
                assertTrue(target.exists(), "Output file missing");
            }
            catch (EncoderException ex)
            {
                throw new AssertionError("Unexpected exception in encoder", ex);
            }
        };
        
        Thread thread = new Thread(task);
        thread.start();
        TimeUnit.MILLISECONDS.sleep(100);
        encoder.abortEncoding();

**Comment by goxr3plus** [comment]
O MY GOD you did it ♥ sorry for bot writing some code on it, just found a new work and 0 time even to eat.

I lovr this library and you for being do active!!!!
